app/components/post_draft/post_draft.js

```diff
@@ -375,14 +476,34 @@ export default class PostDraft extends PureComponent {
 
     sendMessage = () => {
         const value = this.input.current?.getValue() || '';
-        const {enableConfirmNotificationsToChannel, membersCount, useChannelMentions} = this.props;
+        const {enableConfirmNotificationsToChannel, membersCount, useGroupMentions, useChannelMentions, channelMemberCountsByGroup} = this.props;
         const notificationsToChannel = enableConfirmNotificationsToChannel && useChannelMentions;
+        const notificationsToGroups = enableConfirmNotificationsToChannel && useGroupMentions;
         const toAllOrChannel = this.textContainsAtAllAtChannel(value);
+        const groupMentions = this.groupsMentionedInText(value);
 
         if (value.indexOf('/') === 0) {
             this.sendCommand(value);
         } else if (notificationsToChannel && membersCount > NOTIFY_ALL_MEMBERS && toAllOrChannel) {
             this.showSendToAllOrChannelAlert(membersCount);
+        } else if (notificationsToGroups && !toAllOrChannel && groupMentions.length > 0) {
+            let memberNotifyCount = 0;
+            let channelTimezoneCount = 0;
+            const groupMentionsSet = new Set();
+            groupMentions.
+                forEach((group) => {
+                    const mappedValue = channelMemberCountsByGroup[group.id];
+                    if (mappedValue?.channel_member_count > NOTIFY_ALL_MEMBERS && mappedValue?.channel_member_count > memberNotifyCount) {
+                        memberNotifyCount = mappedValue.channel_member_count;
+                        channelTimezoneCount = mappedValue.channel_member_timezones_count;
+                    }
+                    groupMentionsSet.add(`@${group.name}`);
```

Comment:
As per the 1st question, the designs that UX had was that in the confirmation modal we want to display all the groups that were mentioned however only display the number of users in the group with the largest number of users in the channel. Hence why I am keeping track of the group with the largest number of users in the channel but adding every individual group to the `groupMentionsSet` (so they can be shown in the confirmation modal).

For your second question, this follows displaying the number of users being notified for the group with the largest number of users in that channel so we only want the timezones of those users. So it would look something like 
`You are notifying X users in Y timezones` where the Y timezones is regarding the X users. Trying to find the unique number of timezones and what not was decided to be non performant and a difficult task so we just simplified it by going for the group with the largest number of users and then getting the number of unique timezones for those users. (Hopefully my explanation was clear 😅 )

Reply:
I think then the message would be incorrect because the message says `You are about to notify X users in Y timezones` were Y is the number of different timezones for X users. If X was the max number of users being affected and Y was maximum number of timezones being affected, then that messages wording would be incorrect because it wouldn't be Y timezones for X users, it would have to say something like `You are about to notify X users and Y timezones` or something similar.

Reply:
great conversation.. can you extract this block to its own function ?

Reply:
Done.